### DISCUSSION TITLE
[docs] compositional programming: docs and examples?

### DISCUSSION THREAD

**Discussion by oberstet** [p-org/P#699] [Q&A]
The YT talk https://youtu.be/IQd7RIQEFTQ?t=739 is discussing the following keywords

* `interface`
* `receives`
* `sends`
* `creates`

in the context of compositional programming with P.

I wanted to have a deeper look, but the current docs doesn't seem to discuss or even only reference above.

The closest I could find in the docs is https://p-org.github.io/P/manual/modulesystem/, and (almost) the only P code I found here https://github.com/p-org/P/blob/master/Tst/RegressionTests/Feature5ModuleSystem/Correct/InterfaceTest/interface1.p

-----

Is there any docs and/or examples rgd above new keywords (which make a lot of sense in my eyes)?

**Comment by ankushdesai** [comment]
Hi @oberstet, Thanks for your interest in compositional programming using P. 
There is support for all these keywords in the current version of P but with missing documentation.

We didnt add the detailed documentation yet as the user base for it was still small. 
If you are interested in using it, I will work on it and get it added soon.

In the mean-time, you can try and replicate the example in the overview section of the ModP paper: https://dl.acm.org/doi/abs/10.1145/3276529

Thanks again! If you have a concrete use case, please shoot me a separate DM and we can discuss.
